packages/core/src/delegation/DelegationNode.ts

```diff
@@ -98,52 +239,83 @@ export default class DelegationNode extends DelegationBaseNode
   }
 
   /**
-   * [ASYNC] Fetches the root of this delegation node.
+   * [ASYNC] Stores the delegation node on chain.
    *
-   * @throws [[ERROR_ROOT_NODE_QUERY]] when the rootId could not be queried.
-   * @returns Promise containing the [[DelegationRootNode]] of this delegation node.
+   * @param signature Signature of the delegate to ensure it is done under the delegate's permission.
+   * @returns Promise containing a unsigned SubmittableExtrinsic.
    */
-  public async getRoot(): Promise<DelegationRootNode> {
-    const rootNode = await queryRoot(this.rootId)
-    if (!rootNode) {
-      throw SDKErrors.ERROR_ROOT_NODE_QUERY(this.rootId)
+  public async store(signature?: string): Promise<SubmittableExtrinsic> {
+    if (this.isRoot()) {
+      return storeAsRoot(this)
+      // eslint-disable-next-line no-else-return
+    } else {
+      if (!signature) {
+        throw SDKErrors.ERROR_DELEGATION_SIGNATURE_MISSING
+      }
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return storeAsDelegation(this, signature!)
```

Comment:
I think your in-editor linting may be fooling you here, this should be understood by typescript
```suggestion
      return storeAsDelegation(this, signature)
```

Reply:
Fixed in https://github.com/KILTprotocol/sdk-js/pull/406/commits/8a20ab281b2ba7d57cd26af3a0736060d6fa3309.